astropy/io/fits/column.py

```diff
@@ -2063,14 +2063,14 @@ def _update_field_metrics(self):
 class _VLF(np.ndarray):
     """Variable length field object."""
 
-    def __new__(cls, input, dtype="a"):
+    def __new__(cls, input, dtype="S"):
         """
         Parameters
         ----------
         input
             a sequence of variable-sized elements.
         """
-        if dtype == "a":
+        if dtype == "S":
```

Comment:
There were many `if`-statements that handled this `dtype`. Are all these special cases still needed?

Reply:
I've not looked super carefully, but at least in this case am fairly certain it is OK.

Reply:
I'm not familiar with this part of `astropy`, so I can't answer the question I asked myself, but I will point out that "Is this `if`-statement OK?" and "Is this `if`-statement needed?" are two very different questions.

Reply:
Since this is FITS, I would say make the minimal change necessary here and not overthink it.

Reply:
Yes, please do not go in FITS to remove "unneeded" code, definitely not this close to FF, it bit us all back many many times in the past.

Reply:
I'm fairly sure it is needed, as a special path for strings. I also strongly resisted the urge to try to clean up some of the code...